File: buff/classes/354/pub/sum/scala/Sum.scala
// Scala sum program

object Sum {

   def sum(seq: Array[Int]) : Int = {
      var s=0
      for (v<-seq)
         s+=v
      return s
   }

   def main(args: Array[String]) {
      var seq=Array(5,6,1,8,3,7)
      println(sum(seq))
   }

}

File: buff/classes/354/pub/sum/fortran/sum.f
      program main
C     by default, "sum" is an intrinsic function
      external sum
      integer sum
      integer seq(6)
      data seq/5,6,1,8,3,7/
      print *,sum(seq,6)
      return
      end

      integer function sum(seq,n)
      integer seq(*)
      integer n
      sum=0
      do 10 i=1,n
         sum=sum+seq(i)
 10   continue
      return
      end

